src/main/resources/api.yaml

```diff
@@ -139,3 +161,39 @@ definitions:
       - name
       - description
       - initialCoefficient
+  LocalDateTimePeriodDto:
+    type: object
+    properties:
+      start:
+        type: string
+        format: date-time
+      end:
+        type: string
+        format: date-time
+    required:
+      - start
+      - end
+  SprintDto:
+    type: object
+    properties:
+      id:
+        type: integer
+        format: int64
+      enrollmentPeriod:
+        $ref: '#/definitions/LocalDateTimePeriodDto'
+      durationPeriod:
+        $ref: '#/definitions/LocalDateTimePeriodDto'
+      dateClosed:
+        type: string
+        format: date-time
+      timeBurned:
```

Comment:
Jak interpretować tę wartość?

Reply:
Doskonałe pytanie w sumie. Dodanie `InSeconds` do nazwy nam wystarcza?

Reply:
Aa to nawet nie o nazwę mi chodziło, bo warto nazwy trzymać krótkie, ale można zmienić typ (jeśli się da) lub dodać description. Myślę, że nazwa jest okej :)

Reply:
Ups, a siebie samą przekonałam co do `InSeconds` właśnie

Typu dla `Duration` nie znalazłam

Reply:
No bo chodzi o to, że np. jak masz jakieś pole z datą to też nie nazywasz go `dateClosedAsYearMonthDay`, mimo że to również jest string. Format raczej wynika z kontraktu niż z nazwy pola, tak mi się przynajmniej wydaje.

Reply:
Znaczy data przychodzi jako `LocalDateTime` do nas

Reply:
Nic z nim nie trzeba robić

Reply:
A z `Duration` w Mapperach trzeba wyciągnąć sekundy specjalnie

Reply:
Jasne, ale z punktu widzenia użytkownika API RESTowego oba parametry to są stringi.

Reply:
Nazwę zmiennej w Javie możemy sobie ustawić jako `timeBurnedInSeconds` ale wydaje mi się, że RESTem powino iść `timeBurned`.

Reply:
No ale data to data w jakimś tam formacie (najwyżej w złym), a czas trwania może być w nanosekundach, sekundach, minutach, godzinach

Reply:
Czyli czas trwania też jest w jakimś tam formacie (najwyżej złym).

Reply:
Ale data raczej się wywali, jak podasz źle (poza casem miesiąc <-> dzień pewnie), a duration nigdy

Reply:
Znaczy mogę dać opis, spoko, ale nie wydaje mi się, żeby to była ta sama sytuacja

Reply:
można zaskoczyć wszystkich i przesyłać string w formacie HH:mm:ss xd

Reply:
Data się wywali tylko dlatego, że Swagger generuje kod, który ją waliduje. Dla czasu nie ma poprawnego formatu, ale to wina tego, że Swagger tego nie wspiera. Z punktu widzenia API RESTowego różnicy nie ma żadnej bo w obu przypadkach to tylko string, który ma wartości poprawne oraz niepoprawne.

Reply:
Wiesz, mi to obojętne jak to będzie zrobione, tylko mówię jakie jest moje zdanie ;)

Reply:
Haha, ja tak samo - mi też obojętne

Reply:
> można zaskoczyć wszystkich i przesyłać string w formacie HH:mm:ss xd

To by było spoko, tylko no jest szansa na więcej niż 99:59:59

Reply:
MOŻNA ZROBIĆ ENUMA NA JEDNOSTKĘ

Reply:
(Znaczy nie planuję, ALE MOŻNA)

Reply:
prawdę mówiąc nie widzę powodu, żeby sama nazwa zmiennej mówiła w jakiej to jest jednostce, od tego może być `description`

Ja bym proponował przesyłanie tego w minutach (tak podobno jest domyślnie w jirze)

Reply:
Spoko - `Duration` w Javie jakoś tak się lubi z sekundami i stąd się to wzięło, ale minuty też dadzą radę